table/block_based/filter_policy.cc

```diff
@@ -97,6 +109,24 @@ class XXPH3FilterBitsBuilder : public BuiltinFilterBitsBuilder {
   // For delegating between XXPH3FilterBitsBuilders
   void SwapEntriesWith(XXPH3FilterBitsBuilder* other) {
     std::swap(hash_entries_, other->hash_entries_);
+    if (cache_res_mgr_) {
+      std::swap(hash_entry_cache_res_bucket_handles_,
+                other->hash_entry_cache_res_bucket_handles_);
+    }
+    if (detect_filter_construct_corruption_) {
+      std::swap(hash_entries_checksum_, other->hash_entries_checksum_);
+    }
+    entries_swapped_ = true;
+  }
+
+  void ResetEntries() {
+    hash_entries_.clear();
+    if (cache_res_mgr_) {
+      hash_entry_cache_res_bucket_handles_.clear();
+    }
+    if (detect_filter_construct_corruption_) {
+      hash_entries_checksum_ = 0;
+    }
   }
```

Comment:
Note: 
Seems like a good refactoring to have a Struct `HashEntriesInfo` grouping `hash_entries_ + hash_entry_cache_res_bucket_handles_ + hash_entries_checksum_` and do `HashEntriesInfo.Swap(..)`, `HashEntriesInfo.Reset()`instead.

Reply:
Opened task

Reply:
Done